### PR TITLE
feat: Add support for dynamic table grants 

### PR DESCRIPTION
**Pull request by dgniewek**
Added support for Dynamic Tables using new `snowflake_grant_privileges_to_role` terraform resource

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -110,6 +110,28 @@ variable "view_grants" {
...
+    dynamic_table_name = optional(string)
+    on_future          = optional(bool, false)
+    on_all             = optional(bool, false)
+    all_privileges     = optional(bool)
+    privileges         = optional(list(string), null)
+  }))
+  default = []
+  validation {
+    condition     = alltrue([for grant in var.dynamic_table_grants : !anytrue([grant.privileges == null && grant.all_privileges == null, grant.privileges != null && grant.all_privileges != null])])
+    error_message = "Variable `dynamic_table_grants` fails validation - only one of `privilages` or `all_privilages` can be not null."
```
Comment (mariusz89016) [inline variables.tf:127]:
```suggestion
    error_message = "Variable `dynamic_table_grants` fails validation - only one of `privileges` or `all_privileges` can be not null."
```

Comment (mariusz89016) [reply variables.tf:127]: also wdyt about:
> ... only one of `privileges` or `all_privileges` can be set."

?

**Review by mariusz89016 — APPROVED** [review]

File: variables.tf (hunk 1 above)
Comment (dgniewek) [reply variables.tf]: Second sound better - you're right :)

File: main.tf [hunk 2]
```diff
@@ -100,3 +100,35 @@ resource "snowflake_account_grant" "this" {
 
   with_grant_option = false
 }
+
+resource "snowflake_grant_privileges_to_role" "this" {
```
Comment (jakubigla) [inline main.tf:104]: looking at the `each` directive, this should be named not `this` but `dynamic_tables` something?

**Review by jakubigla — APPROVED** [review]

**Pull request merged**